lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -372,12 +374,26 @@ def add_payment(post, payment, options)
           _, _, recurring_detail_reference = payment.split('#')
           post[:selectedRecurringDetailReference] = recurring_detail_reference
           options[:recurring_contract_type] ||= 'RECURRING'
+        elsif payment.is_a?(Check)
+          add_bank_account(post, payment, options[:billing_address][:country])
```

Comment:
This is a very minor thing, but usually we'll pass the entire options hash into `add_` methods like this.